fix: Update citrus-remote to Citrus version 4

maletic opened the pull request:
This pull request updates `citrus-remote` to be compatible with Citrus version 4. The update addresses several dependency issues and ensures that our framework leverages the latest features and improvements of Citrus.
Key Changes:

1. Updated the Citrus dependency to version 4
2. Modified code and configurations to accommodate breaking changes and new features in Citrus version 4
3. Ensured all tests pass with the new version

Testing:

The following repositories were used to test these changes:

1. Citrus: https://github.com/maletic/citrus/tree/fix-test-classes-not-found-in-jar (contains a fix for citrus test class loading)
2. Citrus-Remote: https://github.com/maletic/citrus-remote/tree/update-citrus-remote-to-newest
3. Citrus-Playground: https://github.com/maletic/citrus-playground/tree/update-citrus-remote

This PR depends on: https://github.com/citrusframework/citrus/pull/1188

maletic committed:
Switch spark-core to updated fork

maletic committed:
Fix imports caused by dependency change

maletic committed:
Increase citrus version

maletic committed:
Attempt to fix code problems caused by new citrus version

maletic committed:
Remove archiver as provided

maletic committed:
Use shade plugin to shade jetty and jakarta servlet API

maletic committed:
Upgrade apache http client to version 5

maletic committed:
Update other dependencies to same version as citrus bom

maletic committed:
Change packages from com.consol.citrus to org.citrusframework

bbortt commented:
> This PR depends on: https://github.com/citrusframework/citrus/pull/1188

has been merged, but requires a "fix" release @christophd

bbortt committed:
ci: add maven wrapper

bbortt committed:
chore: code refactoring using sonar

bbortt commented:
I love how the whole project contains 0 tests 🤣 not the best for a testautomation framework haha

@maletic I've looked it through and cleaned up the code- here's the pr: https://github.com/maletic/citrus-remote/pull/1.

I wonder if we could include https://github.com/maletic/citrus-playground/tree/update-citrus-remote into the pipeline here.

@christophd it would be cool if you could "do that thing with the copyright headers again" (you did it for citrusframework/citrus using intellij idea).

maletic committed:
Merge pull request #1 from bbortt/chore/cleanup

code cleanup

maletic committed:
Rename assemblyConfiguration to assembly for backwards compatibility

bbortt committed:
ci: add sample project

props to @maletic and the original sample repository:
https://github.com/maletic/citrus-playground.

bbortt commented:
@maletic & @turing85 please see https://github.com/maletic/citrus-remote/pull/2. I've added the sample repository of @maletic to the build - if you're ok with that?

I am no maintainer, so @christophd will have to approve the workflow in this PR afterwards.

with the changes or additions mentioned above: /lgtm

turing85 commented:
I am against adding the tests as-is. For one, the quarkus versions are outdated. For another, we could rewrite the tests purely in citrus to not pull-in another framework.

Overall, this is unrelated to the actual change. Just like @christophd 's idea to replace spark/jetty with vert.x, I see this as a separate task that should be handled separately.

bbortt commented:
> Overall, this is unrelated to the actual change.

only partially, I think. I do understand that you want to have the change released, however...
1) the module is not continuously tested as it is now
2) I fear that once this PR is merged, nobody is gonna care about tests anymore
    * (or the move to vert.x for that matter)

note that I've additionally checked and updated all dependencies; there were few breaking changes involved too.

anyway, the changes are there, I've submitted the PR. because of that, I might accept the PR without the sample too. as already said, I've no administrative rights on `citrus-remote`, so @christophd will have to do the last call on this one.

- [x] I've manually tested #6 and confirm that it works

bbortt commented on pom.xml:
```diff
@@ -48,14 +48,14 @@
     <apache.rat.plugin.version>0.13</apache.rat.plugin.version>
 
-    <citrus.version>3.4.0</citrus.version>
+    <citrus.version>4.3.0-SNAPSHOT</citrus.version>
```
```suggestion
    <citrus.version>4.3.1</citrus.version>
```

this however will have to be `4.3.1` before merging!

maletic committed:
Fix plugin not being able to read response content

maletic committed:
Incorporate requested changes regarding version upgrades without sample project

maletic commented:
I increased the versions in the poms to the ones in the PR, so it ready for merging 😄

turing85 commented on citrus-remote-maven-plugin/pom.xml:
```diff
@@ -1,11 +1,11 @@
-    <groupId>com.consol.citrus</groupId>
-    <version>3.5.0-SNAPSHOT</version>
+    <groupId>org.citrusframework</groupId>
+    <version>4.3.0-SNAPSHOT</version>
```
We should set this to `4.3.1.-SNAPSHOT`

bbortt replied:
because you want to sync it with citrus? I think that's a bad idea. what if there's a bugfix in citrus-remote, but not in citrus? versions get out of sync so easily. I think it should just be `4.0.0` (and `-SNAPSHOT`).

bbortt replied:
@christophd I think you'd agree here (before merging)

christophd replied:
I think we should try to keep in sync and yes it may be tedious. I would try to only differ in minor releases when there is a bugfix. As a long-term objective we should bring citrus-remote back to citrus core project

bbortt replied:
ok, if that's the goal :)

turing85 commented on citrus-remote-server/pom.xml:
```diff
@@ -4,15 +4,84 @@
-    <groupId>com.consol.citrus</groupId>
-    <version>3.5.0-SNAPSHOT</version>
+    <groupId>org.citrusframework</groupId>
+    <version>4.3.0-SNAPSHOT</version>
```
We should set this to `4.3.1.-SNAPSHOT`

turing85 commented on pom.xml:
```diff
@@ -2,8 +2,8 @@
-  <groupId>com.consol.citrus</groupId>
-  <version>3.5.0-SNAPSHOT</version>
+  <groupId>org.citrusframework</groupId>
+  <version>4.3.0-SNAPSHOT</version>
```
We should set this to `4.3.1.-SNAPSHOT`

turing85 commented:
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L19

```diff
- <maven.version>3.8.4</maven.version>
+ <maven.version>3.9.8</maven.version>
```

turing85 commented:
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L27C12-L27C18

```diff
- <maven.antrun.plugin.version>3.0.0</maven.antrun.plugin.version>
+ <maven.antrun.plugin.version>3.1.0</maven.antrun.plugin.version>
```

turing85 commented:
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L30

```diff
- <maven.clean.plugin.version>3.0.0</maven.clean.plugin.version>
+ <maven.clean.plugin.version>3.3.1</maven.clean.plugin.version>
```

turing85 commented:
https://github.com/citrusframework/citrus-remote/blob/38d346e40f625d8271099a33ea4af66c8334a263/pom.xml#L31

```diff
- <maven.dependency.plugin.version>3.0.2</maven.dependency.plugin.version>
+ <maven.dependency.plugin.version>3.6.0</maven.dependency.plugin.version>
```

turing85 commented:
I think we should sync the plugins versions before merging...

maletic committed:
Increase plugin versions

maletic committed:
Merge branch 'update-citrus-remote-to-newest' into fix/add-sample-to-build

maletic committed:
Merge pull request #2 from bbortt/fix/add-sample-to-build

add sample project

maletic committed:
Revert "add sample project"

maletic committed:
Merge pull request #3 from maletic/revert-2-fix/add-sample-to-build

Revert "add sample project"

maletic committed:
Increase version

turing85 commented:
@christophd so... we gonna merge?

christophd commented on citrus-remote-maven-plugin/pom.xml:
```diff
@@ -1,11 +1,11 @@
-    <groupId>com.consol.citrus</groupId>
-    <version>3.5.0-SNAPSHOT</version>
+    <groupId>org.citrusframework</groupId>
+    <version>4.3.1-SNAPSHOT</version>
```
There is no `4.3.1-SNAPSHOT`. The snapshot versions always use the next major version. This means you need to use `4.4.0-SNAPSHOT` here

maletic replied:
I see, I just updated it 😃

christophd replied:
Thank you! I think we are ready to merge then 🥳

christophd reviewed (commented):
Many thanks, looks good. Just the SNAPSHOT project version should be fixed

maletic committed:
Changed version to 4.4.0-SNAPSHOT

bbortt commented:
thank you so much @maletic and @turing85 for this huge contribution! and all the patience 😉
